modules/Forum/classes/Forum.php

```diff
@@ -450,6 +449,7 @@ public function updateTopicLatestPosts(int $topic_id): void {
                 'topic_last_user' => $latest_post->post_creator,
             ]);
         } else {
+            // Undefined variable $forum_id, what should we do?
             $this->_db->update('forums', $forum_id, [
```

Comment:
cc @samerton this seems to be from the performance improvement stuff about a month ago. What should be done?

Reply:
It looks like `nl2_posts` has a `forum_id` column, I think if we used that column this would work?

Reply:
Problem is that this is the case where the query returns no results, so I think it'd have to be passed into this function as a parameter

Reply:
Ohh haha, facepalm, sure that makes sense. I will update the pr to do so

Reply:
@samerton can you take another look here when you have a chance?
Everywhere except `spam.php` had access to the `forum_id` of the `topic`, so I allowed `forum_id` to be null in the case of `spam.php`. Seems like a code smell and can be investigated and improved later on:tm:

Reply:
Sounds good, we'll revisit to query for forum_id if not provided based off topic_id